www-src/app/views/device_name_picker.coffee

```diff
@@ -38,17 +38,19 @@ module.exports = class DeviceNamePickerView extends BaseView
 
 
         config = app.loginConfig
-        config.deviceName = device
-
-        config.lastInitState = app.init.currentState
-        $('#btn-save').text t 'registering...'
-        app.replicator.registerRemote config, (err) =>
-            if err?
-                log.error err
-                @displayError t err.message
-            else
-                delete app.loginConfig
-                app.init.trigger 'deviceCreated'
+        if config
+            config.deviceName = device
+
+            config.lastInitState = app.init.currentState
+            $('#btn-save').text t 'registering...'
+            app.replicator.registerRemote config, (err) =>
+                if err?
+                    log.error err
+                    return @displayError t err.message
+                else
+                    delete app.loginConfig
+
+        app.init.trigger 'deviceCreated'
```

Comment:
It sound not safe to trigger the event 'deviceCreated' before the event is efectively created ?